docs/performance/sharding.md

```diff
@@ -1,41 +1,133 @@
 (sharding-guide)=
-
 (sharding-performance)=
 
-# Sharding Performance Guide
+# Sharding performance guide
 
-This document is a sharding best practice guide for CrateDB.
+:::{div} sd-text-muted
+Applying sharding can drastically improve the performance on large datasets.
+:::
 
-A brief recap: CrateDB tables are split into a configured number of shards, and
-then these shards are distributed across the cluster.
+This document is a sharding best practice guide for CrateDB.
+A brief recap: CrateDB tables are split into a configured number of shards.
+These shards are distributed across the cluster to optimize concurrent and
+parallel data processing.
 
-Figuring out how many shards to use for your tables requires you to think about
-the type of data you're processing, the types of queries you're running, and
-the type of hardware you're using.
+Whenever possible, CrateDB will parallelize query workloads and distribute them
+across the whole cluster. The more CPUs this query workload can be distributed
+across, the faster the query will run.
 
-:::{NOTE}
+:::{seealso}
 This guide assumes you know the basics.
+If you are looking for an intro to sharding, see also the
+{ref}`sharding-partitioning` and the
+{ref}`sharding reference <crate-reference:ddl-sharding>` documentation.
+:::
+
+
+## General recommendations
+
+To avoid running your clusters with too many shards or too large shards,
+implement the following guidelines as a rule of thumb:
+
+- Use shard sizes between 10GB and 50GB.
+
+- Keep the number of records on each shard below 200 million.
+
+
+## Sizing considerations
+
+General principles requires careful consideration for cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
 
-If you are looking for an intro to sharding, see {ref}`sharding
-<crate-reference:ddl-sharding>`.
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 3-70 GB of data, with 10-50 GB being the sweet spot for most
+workloads. In large clusters, this often means fewer shards than total CPU cores,
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+Smaller shards also result in reduced Lucene index efficiency, which can adversely
+affect computed search term relevance.
```

Comment:
@seut mentioned:

> That's not true as it stands. The number of segments within a shard affects query performance because more segments have to be viewed.

Reply:
"Segments" are currently not yet part of this discussion / document. Maybe we'd introduce them first? For now, I am deleting this statement. If you can up with a better one, or if you think we should add this sentence verbatim, let me know.

> The number of segments within a shard affects query performance because more segments have to be viewed.

Reply:
Removed per aaf9c5376, but later added the statement at a dedicated section per 874339f3e. Let me know if you think we should expand that, or if you would like to see it deleted altogether.